catalog/plugin/catalog-plugin-auditclassifiedaccess/src/main/java/org/codice/alliance/catalog/plugin/auditclassified/AuditClassifiedAccessPlugin.java

```diff
@@ -0,0 +1,121 @@
+ */
+package org.codice.alliance.catalog.plugin.auditclassified;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.common.audit.SecurityLogger;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import org.apache.commons.lang.StringUtils;
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuditClassifiedAccessPlugin implements PostQueryPlugin {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AuditClassifiedAccessPlugin.class);
+
+  private Map<String, List<String>> classifiedValuesMap = new HashMap<String, List<String>>();
+
+  public QueryResponse process(QueryResponse input)
+      throws PluginExecutionException, StopProcessingException {
+
+    List<Result> results = input.getResults();
+
+    results.stream().map(Result::getMetacard).filter(Objects::nonNull).forEach(this::handleAudits);
+
+    LOGGER.trace("Response went through the Audit Classified Access Plugin");
+    return input;
+  }
+
+  private void handleAudits(Metacard metacard) {
+    boolean isClassified = false;
+    Iterator<Entry<String, List<String>>> it = classifiedValuesMap.entrySet().iterator();
+
+    while (it.hasNext() && !isClassified) {
+      Entry<String, List<String>> entry = it.next();
+      if (hasClassifiedValues(entry.getValue(), entry.getKey(), metacard)) {
+        isClassified = true;
+      }
+    }
+
+    if (isClassified) {
+      auditClassifiedMetacard(metacard.getId());
+    }
+  }
+
+  private List<Serializable> getMetacardAttributeValues(Metacard metacard, String attributeString) {
+    List<Serializable> attributeValues = null;
+    Attribute attribute = metacard.getAttribute(attributeString);
+    if (attribute != null) {
+      attributeValues = attribute.getValues();
+    }
+    return attributeValues;
+  }
+
+  private boolean hasClassifiedValues(
+      List<String> classifiedValues, String attributeString, Metacard metacard) {
+
+    List<Serializable> metacardAttributeValues =
+        getMetacardAttributeValues(metacard, attributeString);
+    if (metacardAttributeValues == null) {
+      return false;
+    }
+
+    for (String classifiedValue : classifiedValues) {
+      classifiedValue = classifiedValue.trim();
+      for (Serializable metacardAttributeValue : metacardAttributeValues) {
+        if (classifiedValue.equals(metacardAttributeValue)
+            && !StringUtils.isEmpty(classifiedValue)) {
+          return true;
+        }

```

Comment:
This could be replaced with
```
return classifiedValues
        .stream()
        .map(String::trim)
        .anyMatch(
            classifiedValue ->
                metacardAttributeValues
                    .stream()
                    .anyMatch(
                        metacardAttributeValue ->
                            classifiedValue.equals(metacardAttributeValue)
                                && !StringUtils.isEmpty(classifiedValue)));
```